core/object/object.h

```diff
@@ -676,6 +676,9 @@ class Object {
 
 	Object(bool p_reference);
 
+	template <bool t_reversed>
+	void _notification_fast(int p_notification);
```

Comment:
I don't think `_notification_fast` is a fitting name. After your PR, there will only be one `notification` implementation, and it's this one. "fast" loses meaning then :)
How about calling it `_notification_templated` instead?